### ISSUE TITLE
PII Leak concern

### ISSUE THREAD

**Issue by ppumkin** [stackify/stackify-api-dotnet#129]
Relating to this line we need to turn this off since PII data is being passed to Stackify which is NOT our sub processor

https://github.com/stackify/stackify-api-dotnet/blob/21e4254af00befd19333b5c5977382623f7a6879/Src/StackifyLib/StackifyError.cs#L158

This can also apply to serialising cookies and would be very good to have options to control the level of data being exported from out application.

**Comment by danatcofo** [comment]
Would a setting override to disable this capture suffice?  

Generally this is the system username that is captured and as such is one of the PII exceptions that is in place as its considered necessary for operation. Given that we would default the setting to allow capture.

**Comment by ppumkin** [comment]
In our case the `HttpContext.Current.User.Identity.Name` is probably now what you expected it to be as we have implemented extensions around identity.

Problem is that this `Name` can contain PII data that is considered exported once the data on the wire leaves the region.. which it can do.

The only time this is a problem when there is an exception and looking at the Stackify code it then try and grab allot of contextual information - which in our case is unnecessary. Identity.Name has no value compared to the Stacktrace.

If you were to look at Open Telemetry the approach there is to minimise risk of exporting sensitive data. For example to the point URL's don't contain formatted URLs for routes, say you have a custom route `/api/{tenant}/records`  that is what you will see in the APM. No other context is used unless you explicitly tell it to or add it in code. 

We are restricted with sub processors and the less data exported the safer in our world.

Options that can be set upfront for all kinds of areas of sensitivity would probably be beneficial to everybody in the age of strict data restrictions.

**Comment by danatcofo** [comment]
Yup, @ppumkin we've got this scheduled for implementation with the next release and I'll be working on this issue.  This is an open source project so we do and have accepted 3rd party PR's as well if you want to submit anything.
